File: documentation/user/en/query/requirements/examples/fetching/hierarchyContentViaReference.cs
EvitaResponse<SealedEntity> entities = evita.QueryCatalog(
	"evita",
	session => session.QuerySealedEntity(
        Query(
        	Collection("Product"),
        	FilterBy(
        		AttributeEquals("code", "amazfit-gtr-3"),
        		EntityLocaleEquals(CultureInfo.GetCultureInfo("en"))
        	),
        	Require(
        		EntityFetch(
        			ReferenceContent(
        				"categories",
        				EntityFetch(
        					AttributeContent("code", "name"),
        					HierarchyContent(
        						EntityFetch(
        							AttributeContent("code", "name")
        						)
        					)
        				)
        			)
        		)
        	)
        )
	)
);

File: documentation/user/en/query/requirements/examples/fetching/hierarchyContentViaReference.java
final EvitaResponse<SealedEntity> entities = evita.queryCatalog(
	"evita",
	session -> {
		return session.querySealedEntity(
			query(
				collection("Product"),
				filterBy(
					attributeEquals("code", "amazfit-gtr-3"),
					entityLocaleEquals(Locale.forLanguageTag("en"))
				),
				require(
					entityFetch(
						referenceContent(
							"categories",
							entityFetch(
								attributeContent("code", "name"),
								hierarchyContent(
									entityFetch(
										attributeContent("code", "name")
									)
								)
							)
						)
					)
				)
			)
		);
	}
);